ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,479 @@
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum RegionLock {
+    None,
+    Japan,
+    USA,
+    Europe,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Charset {
+    JapanUSAEurope,
+    China,
+    Korea,
+    Taiwan,
+}
+
+#[derive(Clone, Debug)]
+pub struct MiiDataOptions {
+    pub is_copying_allowed: bool,
+    pub is_profanity_flag_enabled: bool,
+    pub region_lock: RegionLock,
+    pub charset: Charset,
+}
+
+#[derive(Clone, Debug)]
+pub struct SelectorPosition {
+    pub page_index: u8,
+    pub slot_index: u8,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum OriginConsole {
+    ConsoleWii,
+    ConsoleDSi,
+    Console3DS,
+    ConsoleWiiUSwitch,
+}
+
+#[derive(Clone, Debug)]
+pub struct ConsoleIdentity {
+    pub origin_console: OriginConsole,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MiiSex {
+    Male,
+    Female,
+}
+
+#[derive(Clone, Debug)]
+pub struct Details {
+    pub sex: MiiSex,
+    pub birthday_month: u8,
+    pub birthday_day: u8,
+    pub shirt_color: u8,
+    pub is_favorite: bool,
+}
+
+#[derive(Clone, Debug)]
+pub struct FaceStyle {
+    pub is_sharing_enabled: bool,
+    pub shape: u8,
+    pub skin_color: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct FaceDetails {
+    pub wrinkles: u8,
+    pub makeup: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct HairDetails {
+    pub color: u8,
+    pub is_flipped: bool,
+}
+
+#[derive(Clone, Debug)]
+pub struct EyeDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_scale: u8,
+    pub rotation: u8,
+    pub x_spacing: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct EyebrowDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_scale: u8,
+    pub rotation: u8,
+    pub x_spacing: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct NoseDetails {
+    pub style: u8,
+    pub scale: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MouthDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_scale: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MustacheDetails {
+    pub mouth_y_position: u8,
+    pub mustache_style: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct BeardDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct GlassDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MoleDetails {
+    pub is_enabled: bool,
+    pub scale: u8,
+    pub x_position: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MiiData {
+    pub mii_options: MiiDataOptions,
```

Comment:
I think you should remove all the `mii_` prefixes in the struct's members. Most of the time they will be called directly from the `mii_data.[member]` signature.